src/vm/gchelpers.cpp

```diff
@@ -60,6 +60,45 @@ inline gc_alloc_context* GetThreadAllocContext()
     return & GetThread()->m_alloc_context;
 }
 
+// Checks to see if the given allocation size exceeds the
+// largest object size allowed - if it does, it throws
+// an OutOfMemoryException with a message indicating that
+// the OOM was not from memory pressure but from an object
+// being too large.
+inline void CheckObjectSize(size_t alloc_size)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+    } CONTRACTL_END;
+
+    size_t min_obj_size = sizeof(uint8_t*)   // sync block
```

Comment:
hmm, I don't see how this is related to ALIGNCONST... 

yes, all the sizes defined by the GC that're used by vm should be exposed via the interface so you can add this to gcinterface.h.